src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2092,6 +2092,75 @@ def _process_derived_type_decln(self, parent, decl, visibility_map):
             tsymbol.datatype = UnknownFortranType(str(decl))
             tsymbol.interface = UnknownInterface()
 
+    def _get_partial_datatype(self, node, scope, visibility_map):
+        '''Try to obtain partial datatype information from node by removing
+        any unsupported properties in the declaration. This method
+        assumes that the provided declaration statement only declares
+        a single variable.
```

Comment:
I was going to approve, but I am not sure the last sentence here is necessary (it just picks the datatype of the first symbol to return but this is an implementation detail because all datatypes are the same).

E.g.
```
  code = ("program my_prog\n"
          "  real, dimension(N), target :: var1 = 1, var2 = 2\n"
          "end program my_prog\n")
  psyir = fortran_reader.psyir_from_source(code)
  table = psyir.walk(Routine)[0].symbol_table
  print(table.lookup("var1").datatype.partial_datatype)
  print(table.lookup("var2").datatype.partial_datatype)
```
returns:
```
Array<Scalar<REAL, UNDEFINED>, shape=[Reference[name:'n']]>
Array<Scalar<REAL, UNDEFINED>, shape=[Reference[name:'n']]>
```

Can you remove the sentence and add a test case in `test_get_partial_datatype` and `test_process_declarations_unknownfortrantype` to prove this?

Reply:
As discussed, the method itself only supported a single variable and would not return a partial datatype if any additional variables had an assignment. The example you show above works as it is not calling the method directly and PSyclone only calls the method with a single variable. However, I've updated the method to support multiple variables as that seemed the right thing to do.